backend/src/main/java/ai/giskard/service/DatasetService.java

```diff
@@ -122,4 +124,17 @@ public List<FeatureMetadataDTO> getFeaturesWithDistinctValues(Long datasetId) {
         }).toList();
     }
 
+    @Transactional
+    public Dataset renameDataset(long datasetId, String name) {
+        Dataset dataset = datasetRepository.findById(datasetId)
+            .orElseThrow(() -> new EntityNotFoundException(Entity.DATASET, datasetId));
+
+        if (!SecurityUtils.isCurrentUserAdmin() && !SecurityUtils.isCurrentUserInside(dataset.getProject().getOwner())) {
```

Comment:
this check is essentially the same as `canWriteProject`, let's use the same access validation as for dataset creation, e.g. `ai/giskard/web/rest/controllers/UploadController.java:56`